### ISSUE TITLE
End column too far if accent in string

### ISSUE THREAD

**Issue by usinelogicielle** [PyCQA/pylint#8072]
### Bug description

If my code contains accent from my native language. The endColumn of the issue is wrong. The report is used by another program 
which check the location of the issue. And crash because the end of column is wrong.
If i delete the accent, the endColumn seems to be length - 1 than the real end of column.
The file is in UTF-8.

Here the code : 
```
"""
Classe Database Interrogation
:author: usinelogicielle

:version: 0.1

:date: 2022/09/02
"""


class DatabaseInterrogation:
    def get_username(self):
        """
        Méthode permettant de récupérer le token d'accès à l'API de test
        """
        username = ""
        try:
            while username == "":
                username = input("Entrez votre username (pour accéder à test): ")
        except Exception:
            print("Erreur: Accès non valide veuillez reessayer")

```

### Configuration

```ini
[MASTER]
# Specify a configuration file.
#rcfile=

# Python code to execute, usually for sys.path manipulation such as # pygtk.require().
#init-hook=

# Profiled execution.
profile=no

# Add files or directories to the blacklist. They should be base names, not # paths.
ignore=CVS

# Pickle collected data for later comparisons.
persistent=yes

# List of plugins (as comma separated values of python modules names) to load, 
# usually to register additional checkers.
load-plugins=cnes_checker,pylint.extensions.check_elif


[MESSAGES CONTROL]


# Enable the message, report, category or checker with the given id(s). You can
# either give multiple identifier separated by comma (,) or put this option
# multiple time. See also the "--disable" option for examples.
# 2015/07/09 : C0103 (invalid-name/PY.NAME.Convention), R0801 (duplicate-code/COM.PROJECT.CodeCloning) 
#              and R0915 (too-many-statements/COM.MET.LineOfCode) rules enabled
#              In addition, C0102 and W0141 have to be enabled for bad-names and bad-functions rules
disable=all
enable=R5104,W0703,R5402,R5401,R5105,R0203,R0204,R5403,R5106,R0401,W0102,W9097,E0108,R5103,F0002,W9096,C0204,C0203,E0213,C0122,W9095,W0312,W0406,R5101,W0622,W0621,W0404,W0403,R5201,R5301,R5302,R5102,R0915,W0602,E0602,C0412,W0612,C0113,W0603,E0601,C0411,C0413,C0326

max-bool-expr=5
max-returns=1
max-nested-blocks=5

# Disable the message, report, category or checker with the given id(s). You
# can either give multiple identifiers separated by comma (,) or put this
# option multiple times (only on the command line, not in the configuration
# file where it should appear only once).You can also use "--disable=all" to
# disable everything first and then reenable specific checks. For example, if
# you want to run only the similarities checker, you can use "--disable=all
# --enable=similarities". If you want to run only the classes checker, but have
# no Warning level messages displayed, use"--disable=all --enable=classes
# --disable=W"


[REPORTS]

# Set the output format. Available formats are text, parseable, colorized, msvs
# (visual studio) and html. You can also give a reporter class, eg
# mypackage.mymodule.MyReporterClass.
output-format=text

# Put messages in a separate file for each module / package specified on the
# command line instead of printing them on stdout. Reports (if any) will be
# written in a file name "pylint_global.[txt|html]".
files-output=no

# Tells whether to display a full report or only the messages
reports=yes

# Python expression which should return a note less than 10 (10 is the highest
# note). You have access to the variables errors warning, statement which
# respectively contain the number of errors / warnings messages and the total
# number of statements analyzed. This is used by the global evaluation report
# (RP0004).
evaluation=10.0 - ((float(5 * error + warning + refactor + convention) / statement) * 10)

# Add a comment according to your evaluation note. This is used by the global
# evaluation report (RP0004).
comment=no

# Template used to display messages. This is a python new-style format string
# used to format the message information. See doc for all details
#msg-template=


[BASIC]

# Required attributes for module, separated by a comma
required-attributes=

# List of builtins function names that should not be used, separated by a comma
bad-functions=map,filter,apply,input

# Regular expression which should only match correct module names
module-rgx=(([a-z_][a-z0-9_]*)|([A-Z][a-zA-Z0-9]+))$

# Regular expression which should only match correct module level names
const-rgx=(([A-Z_][A-Z0-9_]*)|(__.*__))$

# Regular expression which should only match correct class names
class-rgx=[A-Z_][a-zA-Z0-9]+$

# Regular expression which should only match correct function names
function-rgx=[a-z_][a-z0-9_]{2,30}$

# Regular expression which should only match correct method names
method-rgx=[a-z_][a-z0-9_]{2,30}$

# Regular expression which should only match correct instance attribute names
attr-rgx=[a-z_][a-z0-9_]{2,30}$

# Regular expression which should only match correct argument names
argument-rgx=[a-z_][a-z0-9_]{2,30}$

# Regular expression which should only match correct variable names
variable-rgx=[a-z_][a-z0-9_]{2,30}$

# Regular expression which should only match correct attribute names in class
# bodies
class-attribute-rgx=([A-Za-z_][A-Za-z0-9_]{2,30}|(__.*__))$

# Regular expression which should only match correct list comprehension /
# generator expression variable names
inlinevar-rgx=[A-Za-z_][A-Za-z0-9_]*$

# Good variable names which should always be accepted, separated by a comma
good-names=i,j,k,ex,Run,_

# Bad variable names which should always be refused, separated by a comma
bad-names=foo,bar,baz,toto,tutu,tata

# Regular expression which should only match function or class names that do
# not require a docstring.
no-docstring-rgx=__.*__

# Minimum line length for functions/classes that require docstrings, shorter
# ones are exempt.
docstring-min-length=-1


[FORMAT]

# Maximum number of characters on a single line.
max-line-length=100

# Regexp for a line that is allowed to be longer than the limit.
ignore-long-lines=^\s*(# )?<?https?://\S+>?$

# Allow the body of an if to be on the same line as the test if there is no
# else.
single-line-if-stmt=no

# List of optional constructs for which whitespace checking is disabled
no-space-check=trailing-comma,dict-separator

# Maximum number of lines in a module
max-module-lines=1000

# String used as indentation unit. This is usually " " (4 spaces) or "\t" (1
# tab).
indent-string='    '


[SIMILARITIES]

# Minimum lines number of a similarity.
min-similarity-lines=4

# Ignore comments when computing similarities.
ignore-comments=yes

# Ignore docstrings when computing similarities.
ignore-docstrings=yes

# Ignore imports when computing similarities.
ignore-imports=no


[VARIABLES]

# Tells whether we should check for unused import in __init__ files.
init-import=no

# A regular expression matching the beginning of the name of dummy variables
# (i.e. not used).
dummy-variables-rgx=_$|dummy

# List of additional names supposed to be defined in builtins. Remember that
# you should avoid to define new builtins when possible.
additional-builtins=


[TYPECHECK]

# Tells whether missing members accessed in mixin class should be ignored. A
# mixin class is detected if its name ends with "mixin" (case insensitive).
ignore-mixin-members=yes

# List of classes names for which member attributes should not be checked
# (useful for classes with attributes dynamically set).
ignored-classes=SQLObject

# When zope mode is activated, add a predefined set of Zope acquired attributes
# to generated-members.
zope=no

# List of members which are set dynamically and missed by pylint inference
# system, and so shouldn't trigger E0201 when accessed. Python regular
# expressions are accepted.
generated-members=REQUEST,acl_users,aq_parent


[MISCELLANEOUS]

# List of note tags to take in consideration, separated by a comma.
notes=FIXME,XXX,TODO


[CLASSES]

# List of interface methods to ignore, separated by a comma. This is used for
# instance to not check methods defines in Zope's Interface base class.
ignore-iface-methods=isImplementedBy,deferred,extends,names,namesAndDescriptions,queryDescriptionFor,getBases,getDescriptionFor,getDoc,getName,getTaggedValue,getTaggedValueTags,isEqualOrExtendedBy,setTaggedValue,isImplementedByInstancesOf,adaptWith,is_implemented_by

# List of method names used to declare (i.e. assign) instance attributes.
defining-attr-methods=__init__,__new__,setUp

# List of valid names for the first argument in a class method.
valid-classmethod-first-arg=cls

# List of valid names for the first argument in a metaclass class method.
valid-metaclass-classmethod-first-arg=mcs


[DESIGN]

# Maximum number of arguments for function / method
max-args=5

# Argument names that match this expression will be ignored. Default to name
# with leading underscore
ignored-argument-names=_.*

# Maximum number of locals for function / method body
max-locals=15

# Maximum number of return / yield for function / method body
max-returns=1

# Maximum number of branch for function / method body
max-branches=12

# Maximum number of statements in function / method body
max-statements=80

# Add for RNC CNES
accept-no-param-doc=n
max-mccabe-number=15
max-simplified-mccabe-number=12
max-nested-blocks=6
min-func-comments-ratio=20
min-module-comments-ratio=20
min-func-size-to-check-comments=10
max-bool-expr=5
max-decorators=5

# Maximum number of parents for a class (see R0901).
max-parents=7

# Maximum number of attributes for a class (see R0902).
max-attributes=7

# Minimum number of public methods for a class (see R0903).
min-public-methods=2

# Maximum number of public methods for a class (see R0904).
max-public-methods=20


[IMPORTS]

# Deprecated modules which should not be used, separated by a comma
deprecated-modules=regsub,TERMIOS,Bastion,rexec

# Create a graph of every (i.e. internal and external) dependencies in the
# given file (report RP0402 must not be disabled)
import-graph=

# Create a graph of external dependencies in the given file (report RP0402 must
# not be disabled)
ext-import-graph=

# Create a graph of internal dependencies in the given file (report RP0402 must
# not be disabled)
int-import-graph=


[EXCEPTIONS]

# Exceptions that will emit a warning when being caught. Defaults to
# "Exception"
overgeneral-exceptions=Exception

output-format=json
```

### Command used

```shell
pylint example.py --rcfile=pylintrc --exit-zero
```

### Pylint output

```shell
[
    {
        "type": "warning",
        "module": "example",
        "obj": "DatabaseInterrogation.get_username",
        "line": 20,
        "column": 15,
        "endLine": 20,
        "endColumn": 24,
        "path": "example.py",
        "symbol": "broad-except",
        "message": "Catching too general exception Exception",
        "message-id": "W0703"
    },
    {
        "type": "refactor",
        "module": "example",
        "obj": "DatabaseInterrogation.get_username",
        "line": 18,
        "column": 12,
        "endLine": 19,
        "endColumn": 83,
        "path": "example.py",
        "symbol": "bad-exit-condition",
        "message": "Exit condition based on equality or difference",
        "message-id": "R5103"
    }
]
```

### Expected behavior

I'm expecting that endColumn should be right if there are accent. 
It seems that if there are accents, pylint counts 2 characters instead of one.
It is visible in the R5103 issue. The endColumn is one character superior to the real end of line

```json
[
    {
        "type": "warning",
        "module": "example",
        "obj": "DatabaseInterrogation.get_username",
        "line": 20,
        "column": 15,
        "endLine": 20,
        "endColumn": 26,
        "path": "colibri/frameworks/database/example.py",
        "symbol": "broad-except",
        "message": "Catching too general exception Exception",
        "message-id": "W0703"
    },
    {
        "type": "refactor",
        "module": "example",
        "obj": "DatabaseInterrogation.get_username",
        "line": 18,
        "column": 12,
        "endLine": 19,
        "endColumn": 82,
        "path": "colibri/frameworks/database/example.py",
        "symbol": "bad-exit-condition",
        "message": "Exit condition based on equality or difference",
        "message-id": "R5103"
    }
]
```

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.8.4 (default, Apr 15 2022, 12:14:44)
[GCC 10.2.0]
```

### OS / Environment

CentOS Linux release 7.9.2009 (Core)

### Additional dependencies

affine==2.1.0
alabaster==0.7.12
argon2-cffi==20.1.0
args==0.1.0
asgiref==3.2.7
asn1crypto==0.24.0
astroid==2.3.3
astropy==4.0.1.post1
async-generator==1.10
atomicwrites==1.3.0
attrs==20.3.0
Babel==2.7.0
backcall==0.1.0
bcrypt==3.1.4
bitstring==3.1.5
bleach==3.1.0
bokeh==1.3.4
Bottleneck==1.2.1
CacheControl==0.12.11
cached-property==1.5.2
cachy==0.3.0
certifi==2020.6.20
cffi==1.14.3
cftime==1.0.3.4
chardet==3.0.4
charset-normalizer==2.1.1
cleo==1.0.0a5
click==7.1.2
click-plugins==1.0.4
cligj==0.5.0
clint==0.5.1
cloudpickle==1.2.1
cnes-pylint-extension==6.0.0
colorama==0.4.1
configparser==3.5.1
constantly @ file:///softs/rh7/spack_install/stage/logiciels/spack-stage-py-constantly-15.1.0-ulpwwoztmes77doeflw7jn56sokdkngk/spack-src
coverage @ file:///softs/rh7/spack_install/stage/logiciels/spack-stage-py-coverage-6.3.1-qo6p5yjdkvuqo2os7ebew2soee2s6avm/spack-src
crashtest==0.3.1
cryptography==2.7
cvxopt==1.2.5
cycler==0.10.0
Cython==0.29.21
d2to1==0.2.12.post1
dask==2.16.0
dask-jobqueue @ file:///softs/rh7/spack_install/stage/logiciels/spack-stage-py-dask-jobqueue-0.7.1-24wnflfdhz64duadetrkekb7jov4f6cs/spack-src
deap==1.3.1
decorator==4.4.2
defusedxml==0.6.0
descartes==1.1.0
devlib==1.2.0
dill==0.3.1.dev0
distlib==0.3.6
distributed==2.10.0
Django==3.0.5
docopt==0.6.2
docutils==0.15.2
dulwich==0.20.46
ecdsa==0.15
elasticsearch==7.5.1
entrypoints==0.3
epydoc==3.0.1
filelock==3.8.0
Fiona==1.8.6
fsspec==0.7.3
func-timeout==4.3.5
future==0.18.2
geneticalgorithm==1.0.1
graphviz==0.13.2
h5py==2.10.0
HeapDict==1.0.1
html5lib==1.1
idna==2.8
imageio==2.5.0
imagesize==1.1.0
importlib-metadata==4.12.0
importlib-resources==5.9.0
ipykernel==5.3.4
ipyparallel==6.3.0
ipython==7.18.1
ipython_genutils==0.2.0
ipywidgets==7.5.1
isort==4.3.20
jaraco.classes==3.2.2
jedi==0.13.3
jeepney==0.4.3
Jinja2==3.0.3
joblib==0.14.0
jsonschema==3.2.0
jupyter-client==6.1.7
jupyter-core==4.6.3
jupyterlab-pygments==0.1.1
keyring==23.9.1
kiwisolver==1.1.0
lazy-object-proxy==1.4.3
llvmlite==0.34.0
locket==0.2.0
lockfile==0.12.2
logilab-common==1.4.2
Louie==2.0
lxml==4.5.2
MarkupSafe==2.0.1
matplotlib==3.3.3
mccabe==0.6.1
mistune==0.8.4
mock==3.0.5
more-itertools==7.2.0
mpi4py==3.0.3
mpmath==1.1.0
msgpack==1.0.0
munch==2.2.0
natsort==5.2.0
nbclient==0.5.0
nbconvert==6.0.1
nbformat==5.0.7
nest-asyncio==1.4.0
netCDF4==1.5.3
networkx==2.4
nose==1.3.7
notebook==6.1.4
numba==0.50.1
numexpr==2.7.3
numpy==1.19.4
numpydoc==0.6.0
packaging==19.2
pandas==1.1.4
pandocfilters==1.4.2
paramiko==2.7.1
parso==0.6.1
partd==1.1.0
patsy==0.5.1
pbr==2.0.0
pexpect==4.7.0
pickleshare==0.7.5
Pillow==7.2.0
Pint==0.0.0
pkgconfig==1.5.1
pkginfo==1.8.3
pkgutil_resolve_name==1.3.10
platformdirs==2.5.2
pluggy==0.13.0
ply==3.11
poetry==1.2.0
poetry-core==1.1.0
poetry-plugin-export==1.0.6
progressbar2==3.50.1
prometheus-client==0.7.1
prompt-toolkit==2.0.9
psutil==5.7.2
psycopg2==2.8.6
ptyprocess==0.6.0
py==1.8.0
Py6S==1.8.0
pyasn1==0.4.6
pybind11==2.9.0
pycodestyle==2.6.0
pycparser==2.20
pycrypto==2.6.1
pyface==6.1.2
pyflakes==2.2.0
Pygments==2.6.1
pykml==0.1.3
pylev==1.4.0
pylint==2.3.1
pylint-plugin-utils==0.7
pylint-sonarjson==1.0.6
PyNaCl==1.4.0
pyparsing==2.4.2
pyproj==2.6.0
PyQt5==5.13.1
PyQt5_sip==4.19.21
pyrsistent==0.15.7
pyserial==3.1.1
pyshp==2.1.0
PySide2 @ file:///softs/rh7/spack_install/stage/logiciels/spack-stage-py-pyside2-artifactory-5.15.2-hglvozv73brbf7v4iv3bsrt3iouhrgua/spack-src/PySide2-5.15.2-5.15.2-cp35.cp36.cp37.cp38.cp39-abi3-manylinux1_x86_64.whl
pysolar==0.8
pytest==5.3.4
pytest-runner==5.1
python-dateutil==2.8.0
python-utils==2.4.0
pytz==2020.1
PyWavelets==1.1.1
PyYAML==5.3.1
pyzmq==18.1.0
QtAwesome==0.4.1
qtconsole==4.5.1
QtPy==1.7.1
rasterio==1.1.5
requests==2.24.0
requests-toolbelt==0.9.1
rope==0.11.0
scikit-image==0.17.2
scikit-learn==0.23.2
scikit-optimize==0.5.2
scipy==1.5.4
seaborn==0.9.0
SecretStorage==3.1.2
Send2Trash==1.5.0
setuptools-scm==4.1.2
Shapely==1.7.1
shellingham==1.5.0
shiboken2 @ file:///softs/rh7/spack_install/stage/logiciels/spack-stage-py-shiboken2-5.15.2-iuylw2zwbwylw6pwapkjbpo4klchkgns/spack-src/shiboken2-5.15.2-5.15.2-cp35.cp36.cp37.cp38.cp39-abi3-manylinux1_x86_64.whl
simplejson==3.16.0
simplekml==1.3.1
six==1.14.0
snowballstemmer==2.0.0
snuggs==1.4.1
sortedcontainers==2.1.0
Sphinx==3.2.0
sphinxcontrib-applehelp==1.0.1
sphinxcontrib-devhelp==1.0.1
sphinxcontrib-htmlhelp==1.0.2
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.2
sphinxcontrib-serializinghtml==1.1.3
sphinxcontrib-websupport==1.1.2
sqlparse==0.3.1
statsmodels==0.10.2
stevedore==1.28.0
svgwrite==1.1.12
sympy==1.4
tables==3.6.1
tblib==1.6.0
terminado==0.8.3
testpath==0.4.2
texttable==1.6.1
threadpoolctl==2.0.0
tifffile==2020.10.1
tomli==2.0.1
tomlkit==0.11.4
toolz==0.9.0
tornado==6.0.3
tox==3.26.0
tqdm==4.45.0
traitlets==5.0.4
traits==6.0.0
traitsui==6.1.3
Twisted==15.4.0
typing_extensions==4.3.0
urllib3==1.25.6
virtualenv==16.7.6
virtualenv-clone==0.2.6
virtualenvwrapper==4.8.2
wcwidth==0.1.7
webencodings==0.5.1
widgetsnbextension==3.5.1
wlauto==3.2.0
wrapt==1.11.2
xarray==0.14.0
xlrd==0.9.4
xlwt==1.3.0
zict==1.0.0
zipp==3.8.1
zope.interface==5.1.0

**Comment by Pierre-Sassoulas** [comment]
Thank you for the report. We're not calculating the ``end_column`` our-self, we're using the builtin ast module. So this is an upstream bug in ``ast`` imo.

**Comment by teake** [comment]
Duplicate of https://github.com/PyCQA/astroid/issues/1744.

**Comment by teake** [comment]
@Pierre-Sassoulas Is there an upstream bug report?

**Comment by Pierre-Sassoulas** [comment]
I did not search for it, so maybe.
